src/components/segmentedControl/index.tsx

```diff
@@ -110,7 +110,7 @@ const SegmentedControl = (props: SegmentedControlProps) => {
       updateSelectedSegment(index);
     }
   },
-  [onChangeIndex, selectedSegment, updateSelectedSegment]);
+  [onChangeIndex, updateSelectedSegment]);
```

Comment:
Are you sure that's correct? 
You are using selectedSegment in this callback, removing it can lead to bugs

Reply:
You are right, but actually, I think the condition is not required at all.. 
Fixed :)

Reply:
I think the condition is required (: 
It meant to prevent calling the onChangeIndex callback in case the user press twice at the same segment. 
Now when pressing twice on the same index I still get this callback 😬

Reply:
Actually, IMO the user should handle this (maybe the prop name is misleading and `onSegmentPress` is better).
Because the user might want a callback for any press and we should support it.
Let me know what you think, 'cause I'm not sure what is better :)

Reply:
We can add a separate prop for onPress though I'm not sure it's needed. 
If you think about it, product wise the user wants to be notified on index change, that's the purpose of the component, to change between different segments.
Our goal is to provide an API that is useful for the user